python/ray/dataframe/index_metadata.py

```diff
@@ -272,76 +226,61 @@ def squeeze(self, partition, index_within_partition):
                            'index_within_partition'] -= 1
 
     def copy(self):
-        return _IndexMetadata(coord_df_oid=self._coord_df,
-                              lengths_oid=self._lengths)
+        # TODO: Investigate copy-on-write wrapper for metadata objects
+        coord_df_copy = self._coord_df_cache
+        if not isinstance(self._coord_df_cache, ray.local_scheduler.ObjectID):
+            coord_df_copy = self._coord_df_cache.copy()
 
+        lengths_copy = self._lengths_cache
+        if not isinstance(self._lengths_cache, ray.local_scheduler.ObjectID):
+            lengths_copy = self._lengths_cache.copy()
 
-class _WrappingIndexMetadata(_IndexMetadata):
-    """IndexMetadata implementation for index across a non-partitioned axis.
-    This implementation assumes the underlying index lies across one partition.
-    """
+        index_copy = self._index_cache
+        if self._index_cache is not None:
+            index_copy = self._index_cache.copy()
 
-    def __init__(self, index):
-        """Inits a IndexMetadata from Pandas Index only.
+        return _IndexMetadata(index=index_copy,
+                              coord_df_oid=coord_df_copy,
+                              lengths_oid=lengths_copy)
 
-        Args:
-            index (pd.Index): Index to wrap.
+    def __getitem__(self, key):
```

Comment:
Could you write a comment about how to use `__getitem__`? We have had people using it in incorrect ways and trying to make it work for them, so this way hopefully we can avoid that.